src/api.ts

```diff
@@ -539,39 +568,127 @@ function createApi({
       const offlineStyle: OfflineStyle = {
         ...(await uncompositeStyle(style)),
         id: styleId,
-        sources: await createOfflineSources(style.sources),
+        sources: await createOfflineSources({
+          accessToken,
+          styleId,
+          sources: style.sources,
+        }),
       }
 
       db.prepare(
         'INSERT INTO Style (id, stylejson) VALUES (:id, :stylejson)'
       ).run({ id: styleId, stylejson: JSON.stringify(offlineStyle) })
 
+      // Create the records in the joins table for TilesetsOnStyles
+      // TODO: How to insert multiple records at once?
+      Object.values(offlineStyle.sources).forEach((offlineSource) => {
+        db.prepare<{ tilesetId: string; styleId: string }>(
+          'INSERT INTO TilesetsOnStyles (tilesetId, styleId) VALUES (:tilesetId, :styleId)'
+        ).run({ tilesetId: offlineSource.tilesetId, styleId })
+      })
+
```

Comment:
Is this in order to do a reverse lookup from tileset -> style that references tileset?
A suggestion: don't store the sources property in the db, instead store a JSON column with an array of tilesetIds. Then when getting a style you can re-compose the style property (the type is stored in the tilejson). Would require updating the references to the source id (which are the keys of the sources property) to be the tileset id.
Then you can do an easy query of styles where the sources column includes the tilesetid. Doesn't even matter if it's a full table scan since performance will not be an issue here.

Reply:
i _think_ i have a high level understanding of what you're explaining, but this is a mouthful to process - might be easier if we go through this together 😅

Reply:
I made a change that partially implements your suggestion: https://github.com/digidem/mapeo-mapserver/pull/13/commits/768dce091276ab40bb38d56309a350c1f1b8ad3f

It does the following:

1. Updates each source id in the style's `sources` property to become the tileset id that's used for it
2. Updates the style's `layers` property so that each layer's `source` field (if it exists) is updated to be the corresponding tileset id that replaced the source id

Wasn't entirely sure about the purpose of storing a db field for the tileset ids on the Style table and omitting the source from the saved stylejson. 

> Is this in order to do a reverse lookup from tileset -> style that references tileset?

Yeah that's the idea (and vice versa). For example, getting the tilesets that a style references (based on its `sources` property) would look like this:

```js
// Get Tilesets that a Style references
db.prepare(
  'SELECT * FROM Tileset WHERE id IN (SELECT tilesetId FROM TilesetsOnStyles WHERE styleId = :styleId)'
).all(someStyleId)

// Get Styles that reference a given Tileset
db.prepare(
  'SELECT * FROM Style WHERE id IN (SELECT styleId FROM TilesetsOnStyles WHERE tilesetId = :tilesetId)'
).all(someTilesetId)
```